lib/Mojolicious/Guides/Growing.pod

```diff
@@ -0,0 +1,776 @@
+why L<Mojolicious::Lite> applications don't need more than a single file.
+
+  myapp.pl   # Templates and even static files can be inlined
+
+Full L<Mojolicious> applications on the other hand are much closer to a well organized CPAN distribution to maximize
+maintainability.
+
+  myapp                            # Application directory
+  |- script                        # Script directory
+  |  +- my_app                     # Application script
+  |- lib                           # Library directory
+  |  |- MyApp.pm                   # Application class
+  |  +- MyApp                      # Application namespace
+  |     +- Controller              # Controller namespace
+  |        +- Example.pm           # Controller class
+  |- my_app.yml                    # Configuration file
+  |- t                             # Test directory
+  |  +- basic.t                    # Random test
+  |- log                           # Log directory
+  |  +- development.log            # Development mode log file
+  |- public                        # Static file directory (served automatically)
+  |   |-- assets                   # Static assets created by bundlers
+  |   |   `-- *generated assets*
+  |   +- index.html                # Static HTML file
+  +- templates                     # Template directory
+     |- layouts                    # Template directory for layouts
+     |  +- default.html.ep         # Layout template
+     +- example                    # Template directory for "Example" controller
+        +- welcome.html.ep         # Template for "welcome" action
+
+Both application skeletons can be automatically generated with the commands
+L<Mojolicious::Command::Author::generate::lite_app> and L<Mojolicious::Command::Author::generate::app>.
+
+  $ mojo generate lite-app myapp.pl
+  $ mojo generate app MyApp
+
+Feature-wise both are almost equal, the only real differences are organizational, so each one can be gradually
+transformed into the other.
+
+=head2 Foundation
+
+We start our new application with a single executable Perl script.
+
+  $ mkdir myapp
+  $ cd myapp
+  $ touch myapp.pl
+  $ chmod 744 myapp.pl
+
+This will be the foundation for our login manager example application.
+
+  #!/usr/bin/env perl
+  use Mojolicious::Lite -signatures;
+
+  get '/' => sub ($c) {
+    $c->render(text => 'Hello World!');
+  };
+
+  app->start;
+
+The built-in development web server makes working on your application a lot of fun thanks to automatic reloading.
+
+  $ morbo ./myapp.pl
+  Web application available at http://127.0.0.1:3000
+
+Just save your changes and they will be automatically in effect the next time you refresh your browser.
+
+=head2 A bird's-eye view
+
+It all starts with an HTTP request like this, sent by your browser.
+
+  GET / HTTP/1.1
+  Host: localhost:3000
+
+Once the request has been received by the web server through the event loop, it will be passed on to L<Mojolicious>,
+where it will be handled in a few simple steps.
```

Comment:
It would be super cool if each step could link to the line of code that does the step -- it would really help for better understanding Mojolicious internals which would really help with debugging apps.